### PR TITLE
Hw 03

### PR DESCRIPTION
**Pull request by AliokY**
Home work №3

### CODE REVIEW

File: HW_3/HW03.Alphabet/HW03.Alphabet/Program.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace HW03.Alphabet
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            for (int i = 122; 97 <= i; i--)
```
Comment (VadzimPapko) [inline HW_3/HW03.Alphabet/HW03.Alphabet/Program.cs:9]: 122, 97 - these are magic numbers

File: HW_3/HW03.Alphabet/HW03.Alphabet/Program.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+            {
+                Console.Write(Convert.ToChar(i) + " ");
+            }
+            Console.ReadKey();
+        }
+    }
+}
```
Comment (VadzimPapko) [inline HW_3/HW03.Alphabet/HW03.Alphabet/Program.cs:16]: in general, correct

File: HW_3/HW03.Birthday/HW03.Birthday/Program.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+            Console.WriteLine("Please, enter current date in the format: yyyy-mm-dd");
+            string currentDateInp = Console.ReadLine();
+
+            DateTime bornDate = DateTime.Parse(bornDateInp);
+            DateTime currentDate = DateTime.Parse(currentDateInp);
+
+            DateTime span = new DateTime((currentDate - bornDate).Ticks);
+
+            int age = bornDate.Month == currentDate.Month ? (span.Year) : (span.Year - 1);
```
Comment (VadzimPapko) [inline HW_3/HW03.Birthday/HW03.Birthday/Program.cs:20]: what about if 
`bornDate.Month > currentDate.Month` ?

File: HW_3/HW03.Birthday/HW03.Birthday/Program.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Please, enter the date of birth in the format: yyyy-mm-dd");
+            string bornDateInp = Console.ReadLine();
+
+            Console.WriteLine("Please, enter current date in the format: yyyy-mm-dd");
+            string currentDateInp = Console.ReadLine();
+
+            DateTime bornDate = DateTime.Parse(bornDateInp);
+            DateTime currentDate = DateTime.Parse(currentDateInp);
```
Comment (VadzimPapko) [inline HW_3/HW03.Birthday/HW03.Birthday/Program.cs:16]: good decision!

File: HW_3/HW03.Calculator/HW03.Calculator/Calculator.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+{
+    class Calculator
+    {
+        internal int AddNumbers(int num1, int num2)
+        {
+            int result = num1 + num2;
+
+            return result;
+        }
+        internal int SubtractNumber(int num1, int num2)
```
Comment (VadzimPapko) [inline HW_3/HW03.Calculator/HW03.Calculator/Calculator.cs:13]: put one empty line between methods

File: HW_3/HW03.Calculator/HW03.Calculator/Calculator.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            return result;
+        }
+        internal double CalculateCircleArea(double rad)
+        {
+            double circleArea = Math.PI * Math.Pow(rad, 2);
+
+            return circleArea;
+        }
+    }
+}
```
Comment (VadzimPapko) [inline HW_3/HW03.Calculator/HW03.Calculator/Calculator.cs:44]: correct

File: HW_3/HW03.Calculator/HW03.Calculator/Program.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+            Console.WriteLine("Enter number for calculate circle area");
+            double rad = double.Parse(Console.ReadLine());
+            double circleArea = calculator.CalculateCircleArea(rad);
+            Console.WriteLine($"Division result is {circleArea}");
+
+            Console.ReadLine();
+        }
+    }
+}
```
Comment (VadzimPapko) [inline HW_3/HW03.Calculator/HW03.Calculator/Program.cs:49]: good one

File: HW_3/HW03.Movement/HW03.Movement/Program.cs [hunk 8]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace HW03.Movement
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("enter W, S, A, D to move to the corresponding side");
+            string button = Console.ReadLine().ToUpper();
```
Comment (VadzimPapko) [inline HW_3/HW03.Movement/HW03.Movement/Program.cs:10]: good one

File: HW_3/HW03.Movement/HW03.Movement/Program.cs [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+                    Console.WriteLine("Move right");
+                    break;
+                default:
+                    Console.WriteLine("Movement is not required!");
+                    break;
+            }
+            Console.ReadKey();
+        }
+    }
+}
```
Comment (VadzimPapko) [inline HW_3/HW03.Movement/HW03.Movement/Program.cs:32]: correct

File: HW_3/HW03.Operators/HW03.Operators/Program.cs [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+                }
+                else
+                {
+                    break;
+                }
+            }
+            return operation;
+        }
+
+
```
Comment (VadzimPapko) [inline HW_3/HW03.Operators/HW03.Operators/Program.cs:115]: remove these two lines

File: HW_3/HW03.Operators/HW03.Operators/Program.cs [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+            //Task1();
+
+            //Task2();
+
+            //Task3();
+
+            Task4();
+        }
+    }
+}
```
Comment (VadzimPapko) [inline HW_3/HW03.Operators/HW03.Operators/Program.cs:179]: really good one! Well done!

**Review by VadzimPapko — COMMENTED** [review]
Reviewed. Added some notes.
In general, looks good! Your score is 9

File: HW_3/HW03.Birthday/HW03.Birthday/Program.cs (hunk 3 above)
Comment (AliokY) [reply HW_3/HW03.Birthday/HW03.Birthday/Program.cs:20]: if the current month coincides with the month of birth, then the program considers that the year has passed

**Pull request merged**